test/elixir/test/bulk_docs_test.exs

```diff
@@ -0,0 +1,147 @@
+    # Update just the first doc to create a conflict in subsequent bulk update
+    doc = hd(docs)
+    resp = Couch.put("/#{db}/#{doc._id}", [body: doc])
+    assert resp.status_code == 201
+    # Attempt to delete all docs
+    docs = Enum.map(docs, fn(doc) -> Map.put(doc, :_deleted, true) end)
+    resp = bulk_post(docs, db)
+    # Confirm first doc not updated, and result has no rev field
+    res = hd(resp.body)
+    assert res["id"] == "1" and res["error"] == "conflict"
+    assert Map.get(res, "rev") == nil
+    # Confirm other docs updated normally
+    assert revs_start_with(tl(resp.body), "2-")
+  end
+
+  @tag :with_db
+  test "bulk docs supplies `id` if not provided in doc", ctx do
+    docs = [%{foo: "bar"}]
+    res = hd(bulk_post(docs, ctx[:db_name]).body)
+    assert res["id"]
+    assert res["rev"]
+  end
+
+  @tag :with_db
+  test "bulk docs raises error for `all_or_nothing` option", ctx do
+    opts = [body: %{docs: make_docs(@doc_count), all_or_nothing: true}]
+    resp = Couch.post("/#{ctx[:db_name]}/_bulk_docs", opts)
+    assert resp.status_code == 417
+    assert Enum.all?(resp.body, &(Map.get(&1, "error") == "not_implemented"))
+    expected_reason = "all_or_nothing is not supported"
+    assert Enum.all?(resp.body, &(Map.get(&1, "reason") == expected_reason))
+  end
+
+  @tag :with_db
+  test "bulk docs raises conflict error for combined update & delete", ctx do
+    db = ctx[:db_name]
+    doc = %{_id: "id", val: "val"}
+    resp = Couch.put("/#{db}/#{doc._id}", [body: doc])
+    doc = revise(doc, resp.body)
+    update = %{doc | val: "newval"}
+    delete = Map.put(doc, :_deleted, true)
+    body = bulk_post([update, delete], db).body
+    assert Enum.count(body, &(Map.get(&1, "error") == "conflict")) == 1
+    assert Enum.count(body, &(Map.get(&1, "rev"))) == 1
+  end
+
+  @tag :with_db
+  test "bulk docs raises error for missing `docs` parameter", ctx do
+    docs = [%{foo: "bar"}]
+    resp = Couch.post("/#{ctx[:db_name]}/_bulk_docs", [body: %{doc: docs}])
+    assert_bad_request(resp, "POST body must include `docs` parameter.")
+  end
+
+  @tag :with_db
+  test "bulk docs raises error for invlaid `docs` parameter", ctx do
+    resp = Couch.post(
+      "/#{ctx[:db_name]}/_bulk_docs", [body: %{docs: "foo"}])
+    assert_bad_request(resp, "`docs` parameter must be an array.")
+  end
+
+  @tag :with_db
+  test "bulk docs raises error for invlaid `new_edits` parameter", ctx do
+    opts = [body: %{docs: [], new_edits: 0}]
+    resp = Couch.post("/#{ctx[:db_name]}/_bulk_docs", opts)
+    assert_bad_request(resp, "`new_edits` parameter must be a boolean.")
+  end
+
+  @tag :with_db
+  test "bulk docs emits conflict error for duplicate doc `_id`s", ctx do
+    docs = [
+      %{_id: "0", a: 0},
+      %{_id: "1", a: 1},
+      %{_id: "1", a: 2},
+      %{_id: "3", a: 3}]
+    body = bulk_post(docs, ctx[:db_name]).body
+    assert length(body) == 4
+    assert Enum.count(body, &(Map.get(&1, "error") == "conflict")) == 1
```

Comment:
The original JS test do it differently. It checks that:
- second element has: `id = "1"` and `error = undefined`
- third element has: `error = "conflict"`

Your version of the test would pass if we change implementation of the endpoint to return (for example):
```
[
   %{_id: "0", error: "conflict"},
   %{_id: "1", a: 1},
   %{_id: "1", a: 2},
   %{_id: "3", a: 3}
]
```

I think the test need to be changed to make sure that:
- it is exactly on element with `error = "conflict"` and its `_id` is `1`

Reply:
Yes, you are right! I was trying to be clever, but the original assertions are much stronger. Fixed.